test/test.media.ts

```diff
@@ -142,6 +142,38 @@ describe('Media', () => {
     scope.done();
   });
 
+  // See: https://github.com/googleapis/google-api-nodejs-client/issues/1820
+  it('should post progress for uploads, for APIs with empty requestBody', async () => {
+    const scope = nock(Utils.baseUrl)
+      .post(
+        '/upload/youtube/v3/thumbnails/set?videoId=abc123&uploadType=multipart'
+      )
+      .reply(200);
+    const fileName = path.join(__dirname, '../../test/fixtures/mediabody.txt');
+    const fileSize = fs.statSync(fileName).size;
+    const google = new GoogleApis();
+    const youtube = google.youtube('v3');
+    const progressEvents = new Array<number>();
+    await youtube.thumbnails.set(
+      {
+        videoId: 'abc123',
```

Comment:
This test demonstrates how to perform a multipart upload with an API that doesn't expect a `requestBody`.